### PR TITLE
Query: Clean up query validation for various negative cases

### LINKED ISSUE #15312: QueryRewrite: clean up query validation for various negative cases
We have a number of tests that verify we validate negative inputs correctly. Those are now different in many places. Disabling tests for now and will clean them up once we have a fuller picture of the entire pipeline.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #15312

### CODE REVIEW

File: test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs [hunk 1]
```diff
@@ -3581,9 +3581,7 @@ public virtual async Task Include_collection_and_invalid_navigation_using_string
                     async,
                     ss => ss.Set<Gear>().Include("Reports.Foo")))).Message;
 
-            Assert.Equal(
-                CoreStrings.IncludeBadNavigation("Foo", "Gear"),
-                message);
+            Assert.Contains("Invalid include path 'Reports.Foo', couldn't find navigation for 'Foo'.", message);
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs:3584]: Why not `CoreStrings`?

Comment (smitpatel) [reply test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs:3584]: Warning as error. I could not even find CoreString.Method in designer.cs

Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs:3584]: `CoreStrings.LogInvalidIncludePath`

Comment (smitpatel) [reply test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs:3584]: It is not there.
There is LoggingDefinitions.LogInvalidIncludePath which returns EventDefinitionsBase
CoreResources.LogInvalidIncludePath which takes IDiagnosticsLogger and returns EventDefinition<object, object>

Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs:3584]: CoreResources.LogInvalidIncludePath(new TestLogger<TestLoggingDefinitions>()).GenerateMessage("Reports.Foo", "Foo")

**Pull request merged**